### ISSUE TITLE
All.js should mount a LIVE dashboard that renders survey cards to a user then submits results to the server

### ISSUE THREAD

**Issue by wannabeCitizen** [SFII/scq#38]
Should run gulp task that imports dashboard and cards then mounts the dashboard in place in the HTML

**Comment by antsankov** [comment]
This already seems to be done. Maye a little more clarification? Also, @Bradyta a useful task could be cleaning up the package.json and the gulpfile to remove any unused imports.

**Comment by wannabeCitizen** [comment]
@antsankov, if this were done then we would have a webpage. This is supposed to be a final version of the page dependent on @CarlosHL and @MCChung303. @Bradyta is responsible for making sure he understands the gulp build and that the page's dependencies build and render on request to the backend.

**Comment by Brady76** [comment]
@wannabeCitizen I think @antsankov might be right, after the meeting I looked at the gulpfile which takes 
js_files = [
    './src/static/javascripts/db.js',
    './src/static/javascripts/components/questions.jsx',
    './src/static/javascripts/components/cards.jsx',
    './src/static/javascripts/components/dashboard.jsx',
    './src/static/javascripts/components/welcome.jsx'
]
And concatenates them into the all.js file in ./src/static/dist/
The HTML file mounts the dashboard & cards in src/templates/layouts/standard/body.html on lines 24-26.
I think the web page is the dashboard that runs after running make serve is it not?

**Comment by wannabeCitizen** [comment]
Yes it mounts, but it doesn't run live on any data.

The whole goal of what me, you, carlos, and michael discussed is that these
components should run live on data. Once Carlos and Michael upload their
stuff, your job will be to make sure everything builds and mounts correctly.

If you think it's done that's fine, but know I will be asking you 3 to show
me a fully-working front-end for next week.

This confusion is really concerning me that people don't really have a good
idea of what "working" and "not working" look like. Just because a
component mounts does not mean the dashboard is done.

**Comment by wannabeCitizen** [comment]
@Bradyta @CarlosHL @MCChung303 and @antsankov you all need to get on the same page because there appears to be some confusion around what's "working" and what's not. You guys need to make sure that all the different kinds of cards actually render using live data. @untra and @nicot already have an endpoint for you to test rendering and saving surveys. 

You may think your task is "done" because it renders something, but you guys aren't done until all the cards can mount and you can actually receive and submit surveys. I'm using `origin/master` to run our site. I am not seeing cards or surveys. Are you guys? If so, on what branch?

**Comment by wannabeCitizen** [comment]
Alright, in light of all this confusion I've reworded this @Bradyta. But really your job is to integrate the finalized components and make sure they all mount when receiving .json survey specifications and send the results when you answer a question.

**Comment by Brady76** [comment]
Oh, I knew that it wasn't running on live data yet, it's running on a hard coded sample survey.json file so that's not where the confusion is. I thought that my personal task of having a mounted all.js file created through a gulp task might be completed already, and that it's current implementation might work seamlessly as soon as the dashboard.jsx file is updated with a full page.json handler that sends individual survey.json's down to Card and an AJAX get that fetches that page.json is the first place so we have live data.  So I knew the front-end team wasn't done with development of a live dashboard, I just thought that this issue you assigned me individually involving the gulpfile might be done and you may not have noticed and I wanted to be productive and fair by asking for another task if that was true so I asked Alex who agreed. I could imagine that there might be additional dependencies involved in the gulp file once AJAX is involved but I have no experience so I don't know. I probably should've just came and asked you here immediately to avoid the disconnect, so I'm sorry. Just to make sure we see the same dashboard I'm attaching a screenshot, the cards are again rendered from a hard coded example json.
![dashboard](https://cloud.githubusercontent.com/assets/4028785/11614452/b66d2c2c-9c00-11e5-973e-c69e6080ca45.png)

**Comment by wannabeCitizen** [comment]
Ok yes this makes sense. Sorry for confusion on my end. You definitely can
work on other things but I was imagining you as in charge of merging the
code that Carlos and Michael give you, building it and testing it with the
login flow.

Sorry, I just assumed that whoever was in charge of the build file would
test and merge.

Also, I get a different screen at /dashboard. Is that post login? I
couldn't get the ldap login to work.

**Comment by wannabeCitizen** [comment]
Hope the earlier stuff didn't sound harsh. I just got worried that it's end
of semester and we are having severe miscommunication.

I do agree with @atsankov that if the new files aren't ready to test and
merge, you might as well do something else. Just know you're responsible
for the final build and test of the full dashboard.

I do aplogize if I came off rash, simply concerned about getting on the
same page.

Oh, I knew that it wasn't running on live data yet, it's running on a hard
coded sample survey.json file so that's not where the confusion is. I
thought that my personal task of having a mounted all.js file created
through a gulp task might be completed already, and that it's current
implementation might work seamlessly as soon as the dashboard.jsx file is
updated with a full page.json handler that sends individual survey.json's
down to Card and an AJAX get that fetches that page.json is the first place
so we have live data. So I knew the front-end team wasn't done with
development of a live dashboard, I just thought that this issue you
assigned me individually involving the gulpfile might be done and you may
not have noticed and I wanted to be productive and fair by asking for
another task if that was true so I asked Alex who agreed. I could imagine
that there might be additional dependencies involved in the gulp file once
AJAX is involved but I have no experience so I don't kno w. I probably
should've just came and asked you here immediately to avoid the disconnect,
so I'm sorry. Just to make sure we see the same dashboard I'm attaching a
screenshot, the cards are again rendered from a hard coded example json.
[image: dashboard]
https://cloud.githubusercontent.com/assets/4028785/11614452/b66d2c2c-9c00-11e5-973e-c69e6080ca45.png

**Comment by Brady76** [comment]
Yeah I'm happy to make sure that Carlos and Michael's code merges correctly, it just seems like the coding aspect is practically done already so I asked for a bit more work this week. I don't think I know much at all about the login flow so I'll ask Sam about that and see how much attention that might require, he's actually been jumping in with some front end work. It's not involving login at all, I just do rethinkdb, make serve, then put http://localhost:8000/dashboard into chrome. Is there a different way that I should be running this? It's fine, no hurt feelings here.

**Comment by antsankov** [comment]
I agree that this is a good task for keeping in mind. For now, maybe try to work on https://github.com/SFII/scq/issues/42. We can make sure this task is finished by next week once POST request is done to the server.

**Comment by Brady76** [comment]
So our gulpfile still works and concatenates into all.js which is mounted onto the dashboard, ajax GET and POST are working (I think) but it's impossible to tell until I know the back end is working as well. I hope this is the kind of update you were looking for @antsankov, I think this can't be closed yet because we're not getting and posting live data yet.

**Comment by wannabeCitizen** [comment]
Hey @Bradyta,

I'm seeing an AJAX function getsurvey(), but it's not called anywhere. How
does it get triggered? Also, do you know what parameters your sending?

**Comment by Brady76** [comment]
That's not a function I was ever using, Alex just wrote that out so we could see what ajax might look like I believe.

**Comment by wannabeCitizen** [comment]
Then what leads you to believe the GET and POST are working? Have you done
in-browser network tracking to test that it sends the HTTP requests as
expected?

**Comment by Brady76** [comment]
It's hard to say for sure without knowing if the back end is there or not,
but I have some console.log() unit testing in place and I have 500 errors
in the console which indicate server side issues but as far as I can tell
my console testing indicates I'm sending a JSON object when I click submit
and that it at least tries to fetch live data when the react is mounted.

**Comment by wannabeCitizen** [comment]
As long as its sending with all the parameters and headers as you expect,
then it should be ready. Are you coordinating with carlos, who is supposed
to be finalizing the requests on the survey cards?

**Comment by Brady76** [comment]
Okay then yeah I think I'm there. I talked with him a bit today and he
showed me a way he's been working on to get the initial page and I have yet
to incorporate that but yeah we've been working. We've also been talking
with Sam but our meeting plan fell through the other day.

**Comment by CarlosHL** [comment]
I was with Sam last Friday. We are receiving data that now I have to pass to react, and modify the handler cause is in a different format. This data (the initial) is not passed through Ajax, so currently I am not working with Ajax. I had commented that on my issue.

**Comment by untra** [comment]
@CarlosHL ditto
I'll add that what's missing is the react endpoint to _consume_ a json survey array, and turn the json into the cards. It's the last task before frontend and backend are **100%** connected.

**Comment by Brady76** [comment]
This is done so long as the JSON doesn't unexpectedly not pass @untra 's tests.
